pkg/k8s/node.go

```diff
@@ -38,6 +38,8 @@ func ParseNodeAddressType(k8sAddress slim_corev1.NodeAddressType) (addressing.Ad
 
 // ParseNode parses a kubernetes node to a cilium node
 func ParseNode(k8sNode *slim_corev1.Node, source source.Source) *nodeTypes.Node {
+	var encountered bool
+	addressTypes := make(map[slim_corev1.NodeAddressType]bool)
```

Comment:
This map doesn't need to hold a `bool` and can instead hold an empty struct `struct{}`